programmers/이상한-문자-만들기.js

```diff
@@ -0,0 +1,20 @@
+function solution(s) {
+    let answer = '';
+    let index = 0;
+    for (let i = 0; i < s.length; i++) {
+        if(s[i] === " ") {
+            index = 0;
+            answer += " ";
+        }
+        else {
+            if(index % 2 === 0) {
+                answer += s[i].toUpperCase();
+            }
+            else {
+                answer += s[i].toLowerCase();
+            }
+            index++;
+        }
+    }
+    return answer;
+}
```

Comment:
우선 가장 먼저 왜 for문안에 index = 0을 또 할당하냐면, for문을 반복하는 과정에서 `띄어쓰기(" ")`를 만났을 때 index를 0으로 할당 해야 다음 단어를 만났을 때, index가 0부터 다시 증감해서 `index % 2`를 통해 올바르게 값을 알아낼 수 있음 . 예로들어 for문 안에 index = 0 을 뺴면
t -> 0
r -> 1
y - > 2
" " -> 3
H -> 4
e -> 5
. . . 

이런식으로 index의 값이 `계속 증가`하게 됨. 이럴 경우 index % 2를 했을 때 원하는 값을 도출해 낼 수 없음

t -> 0
r -> 1
y -> 2
" " -> 0
H -> 0
e -> 1
이렇게 `띄어쓰기(" ")`를 만났을 때는 index가 0이 되고 다시 증감해야됨
```suggestion
function solution(s) {
    let answer = '';
    let index = 0;
    
    for (let i = 0; i < s.length; i++) {
        if (s[i] === " ") {
            index = 0;
            answer += " ";
        }
        else {
            if (index % 2 === 0) answer += s[i].toUpperCase();
            else answer += s[i].toLowerCase();
            
            index++;
        }
    }
    return answer;
}
```

Reply:
문자열 말고 배열로도 풀 수 있음 (참고)
```js
function solution(s) {
    const result = [];
    
    let temp = [];
    
    s.toLowerCase().split(" ").map(word => {
        temp = [];
        
        word.split("").map((el, i) => {
            if (i % 2 === 0) temp.push(el.toUpperCase());
            else temp.push(el);
        });
        
        result.push(temp.join(""));
    });
    
    return result.join(" ");
}
```